File: jpa-example/src/main/java/example/jpa/model/Store.java
package example.jpa.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.LinkedHashSet;
import java.util.Set;

@Getter
@Setter
@Entity
public class Store extends SerialId {

    @Version
    private int version;

    @Basic(fetch = FetchType.LAZY)
    private String name;

    @Basic(fetch = FetchType.LAZY)
    private String website;

    @ManyToMany(mappedBy = "stores")
    private Set<Book> books = new LinkedHashSet<>();
}

File: jimmer-example/src/main/kotlin/example/jimmer/model/Store.kt
package example.jimmer.model

import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.ManyToMany
import org.babyfish.jimmer.sql.Version

@Entity
interface Store : SerialId {

    @Version
    val version: Int

    val name: String

    val website: String?

    @ManyToMany(mappedBy = "stores")
    val books: List<Book>
}
